gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/source/extractor/extract/kafka/KafkaExtractorStatsTracker.java

```diff
@@ -398,5 +480,8 @@ public void reset() {
     for (int partitionIdx = 0; partitionIdx < this.partitions.size(); partitionIdx++) {
       resetStartFetchEpochTime(partitionIdx);
     }
+    if (this.observedLagHistogram != null) {
+      this.observedLagHistogram.reset();
```

Comment:
Is the `reset` method reclaim memory ? Would it be safer if destroy the object and re-create next time if reclaim is not happening ?

Reply:
Added benchmark to compare reset vs a new Histogram creation. While both reset and new are cheap, reset is 3x cheaper than new object creation. reset simply keeps the allocated count array as is and zeroes out the array. In general, it would be better to avoid new object creation to avoid GCs/memory fragmentation that can occur over time.

Reply:
+1